File: app/src/main/java/intDef/demo/ClassTest2.java
package intDef.demo;

import com.yy.mobile.whisper.IntDef;
import com.yy.mobile.whisper.StringDef;

/**
 * @author YvesCheung
 * 2019-12-26
 */
public class ClassTest2 {

    private void checkInt(@IntDef({1, 3, 5}) int param) {}

    private void checkIntArray(@IntDef({1, 3, 5}) int... param) {}

    private void checkString(@StringDef({"Hello", "World"}) String param) {}

    public static void main(String[] args) {
        ClassTest2 instance = new ClassTest2();

        instance.checkInt(6);

        instance.checkIntArray(1, 4, 6);

        instance.checkString("Hello World");
    }
}

File: app/src/main/java/usewith/demo/ClassTest2.kt
package usewith.demo

import com.yy.mobile.whisper.UseWith

/**
 * @author YvesCheung
 * 2019-12-19
 */
class ClassTest2 {

    private val instance = SDK()

    private fun init() {
        instance.init()
    }

    private fun unInit() {
        //instance.deInit()
    }

    class SDK {

        @UseWith("deInit")
        fun init() {
            //init sdk
        }

        fun deInit() {
            //Don't forget deInit!!
        }
    }
}

